### DISCUSSION TITLE
How to listen event response Markup.button.callback('Yes', 'yes') on other user

### DISCUSSION THREAD

**Discussion by nodirnavruzov** [telegraf/telegraf#1653] [General]
Hi everyone, i have chat bot when user John want to close chat with user Michael he send to bot command /close then i send to user Michael     await ctx.telegram.sendMessage(chatId, 
      `❗️ The interlocutor wants to close the question, do you mind?` , 
      {
        parse_mode: 'HTML', 
        ...Markup.inlineKeyboard(
          [
            [
                Markup.button.callback('Yes', 'yes'), 
            ],
            [
                Markup.button.callback('No', 'no'), 
            ]
          ]
        )
      }
    ) with options yes or no, how i can listen response to this event? if Michael click to button yes i need clear chat on John. Sorry for my english if there are any mistakes)

**Comment by MKRhere** [comment]
```TS
bot.action("yes", ctx => { /* handle yes */ });
bot.action("no", ctx => { /* handle no */ });
```

**Reply by code-by** [reply]
I try with `Markup.button.callback` on regular (not inline) `keyboard` on my `Scene`, but it not react to button clicks. Shall I use `button.callback` only with `inlineKeyboard`?

**Reply by MKRhere** [reply]
Yes, callback buttons are only supported in inline keyboard.

**Reply by code-by** [reply]
> Yes, callback buttons are only supported in inline keyboard.

so I need to use only `hear` to listen keyboard buttons?

**Reply by MKRhere** [reply]
Correct. It sends as a user message, which you can listen for using `hears`
